Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -698,13 +705,35 @@ - (NSIndexPath *_Nullable)indexPathOfUnreadMessagesIndicator
     return nil;
 }
 
+- (NSIndexPath *_Nullable)indexPathOfMessageOnOpen
+{
+    OWSAssert(self.focusMessageIdOnOpen);
+
+    NSInteger row = 0;
+    for (ConversationViewItem *viewItem in self.viewItems) {
```

Comment:
Are you sure we can't get this in constant time from the mapping?

Reply:
We can get constant time by using the position.